### PR TITLE
chore: fix typos

### PR DESCRIPTION
**Pull request by szepeviktor**
**Changes proposed in this pull request:**

Found few misspellings.

**Reviewers should focus on:**

Please be gentle with me. I am not a Flarum user.

https://github.com/crate-ci/typos

### CODE REVIEW

**Review by SychO9 — CHANGES_REQUESTED** [review]
Thanks for the PR.

But please undo any changes made to migration files, as that'll break existing flarum installations.

**Comment by szepeviktor** [comment]
Okay. Reverting ...

File: framework/core/js/src/common/Application.tsx [hunk 1]
```diff
@@ -293,7 +293,7 @@ export default class Application {
 
         caughtInitializationErrors.push(() =>
           fireApplicationError(
-            extractText(app.translator.trans('core.lib.error.extension_initialiation_failed_message', { extension })),
+            extractText(app.translator.trans('core.lib.error.extension_initialization_failed_message', { extension })),
```
Comment (luceos) [inline framework/core/js/src/common/Application.tsx]: This will break all translations.

File: framework/core/locale/core.yml [hunk 2]
```diff
@@ -734,7 +734,7 @@ core:
       circular_dependencies_message: "Circular dependencies detected: {extensions}. Aborting. Please disable one of the extensions and try again."
       db_error_message: "Database query failed. This may be caused by an incompatibility between an extension and your database driver."
       dependent_extensions_message: "Cannot disable {extension} until the following dependent extensions are disabled: {extensions}"
-      extension_initialiation_failed_message: "{extension} failed to initialize, check the browser console for further information."
+      extension_initialization_failed_message: "{extension} failed to initialize, check the browser console for further information."
```
Comment (luceos) [inline framework/core/locale/core.yml]: This will break all translations.

**Review by luceos — CHANGES_REQUESTED** [review]
I think we should prevent locale strings as they will break all language packs.

In addition, although it makes sense to run a typo package once to create this PR. I'd rather see something like this be made part of our pipeline with the proper folders configured to check.

Opinions?

File: framework/core/locale/core.yml (hunk 2 above)
Comment (szepeviktor) [reply framework/core/locale/core.yml]: Reverted both.
This is `.typos.toml`
```toml
[files]
extend-exclude = [
    ".git/",
    "**/dist/",
    "**/dist-typings/",
]
ignore-hidden = false

[default]
extend-ignore-re = [
    "\\bStichting Flarum\\b",
    "\\(follow\\|ignor\\)",
    "\\bqueue\\.failer\\b",
    "\\[[0-9a-f]{7}\\]",
]

[default.extend-words]
"encrypter" = "encrypter"
"invokable" = "invokable"
"responsable" = "responsable"

[default.extend-identifiers]
routeDatas = "routeDatas"
2018_09_15_041340_add_users_indicies = "2018_09_15_041340_add_users_indicies"
2018_09_15_041828_add_discussions_indicies = "2018_09_15_041828_add_discussions_indicies"
2018_09_29_060444_replace_emoji_shorcuts_with_unicode = "2018_09_29_060444_replace_emoji_shorcuts_with_unicode"
extension_initialiation_failed_message = "extension_initialiation_failed_message"
```

**Comment by DavideIadeluca** [comment]
@luceos I stumbled across https://github.com/codespell-project/actions-codespell which also allows making it part of the pipeline like you were mentioning.

Check out https://github.com/codespell-project/codespell/blob/main/README.rst how it works and see it in action at https://github.com/glowingblue/flarum-framework-2.x/actions/runs/11314056207/job/31463551626. Definitely needs some fine-tuning, though.

Opinions on adopting such a workflow?

**Comment by szepeviktor** [comment]
Typos is a superior product. Also has a built-in GHA action.
Please consider using it. Configuration above!
https://github.com/crate-ci/typos

**Comment by SychO9** [comment]
Imo, the pipeline is already intense as it is, with frontend typings and format checking, + backend tests for various environements. I'd rather see a typo PR before every release, then more failures or auto commits just for typos.

**Review by SychO9 — APPROVED** [review]

**Comment by szepeviktor** [comment]
Glad to contribute.

**Pull request merged**